packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -470,14 +470,15 @@ export default function useAutocomplete(props) {
       previousProps.filteredOptions &&
       previousProps.filteredOptions.length !== filteredOptions.length &&
       (multiple
-        ? previousProps.value.every((val, i) => getOptionLabel(value[i]) === getOptionLabel(val))
-        : getOptionLabel(previousProps.value ?? '') === getOptionLabel(value ?? ''))
+        ? value.length === previousProps.value.length &&
+          previousProps.value.every((val, i) => isOptionEqualToValue(value[i], val))
+        : isOptionEqualToValue(previousProps.value ?? '', value ?? ''))
     ) {
       const previousHighlightedOption = previousProps.filteredOptions[highlightedIndexRef.current];
 
       if (previousHighlightedOption) {
         const previousHighlightedOptionExists = filteredOptions.some((option) => {
-          return getOptionLabel(option) === getOptionLabel(previousHighlightedOption);
+          return isOptionEqualToValue(option, previousHighlightedOption);
```

Comment:
The second argument was meant to be a value, not an option. It's related to #23708 and #31192. Unclear what is right here.

Reply:
Ah, ok. We could use something similar to `optionComparer` from Base's useListbox: https://github.com/mui/material-ui/blob/f800c835562da9cc8fa3f7615b32a8e6cc8b31d8/packages/mui-base/src/ListboxUnstyled/useListbox.ts#L35
But then introducing yet another prop looks like an overkill. I'll leave just the bare minimum to fix the issue here, and we can discuss the approach for changing `getOptionLabel` elsewhere.